File: nim/src/day02.nim
import std/strutils
import std/sugar
import math

proc loadData(): seq[array[0..2, int]] =
  let rows = readFile("../data/day02.txt").strip.splitLines
  let sizes = collect(newSeq):
    for row in rows:
      let splitted = row.split("x")
      [parseInt(splitted[0]), parseInt(splitted[1]), parseInt(splitted[2])]
  return sizes

proc amountOfPaper(l: int, w: int, h: int): int =
  return 10


proc day02_part1(): int =
  let sizes = loadData()
  let amounts = collect(newSeq):
    for dimensions in sizes:
      amountOfPaper(dimensions[0], dimensions[1], dimensions[2])
  return sum(amounts)


when isMainModule:
  echo(day02_part1())

File: python/day02.py
from pathlib import Path


def load_data():
    content = Path('../data/day02.txt').read_text()
    lines = content.split('\n')
    return [
        [int(value) for value in line.split('x')]
        for line in lines
    ]


def amount_of_paper(length: int, width: int, height: int) -> int:
    sides = [
        length * width,
        width * height,
        height * length,
    ]
    return 2 * sum(sides) + min(sides)


def amount_of_ribbon(length: int, width: int, height: int) -> int:
    sides = [
        length + width,
        width + height,
        height + length,
    ]
    return 2 * min(sides) + length * width * height


def main():
    measures = load_data()
    print(sum(amount_of_paper(*measure) for measure in measures))
    print(sum(amount_of_ribbon(*measure) for measure in measures))


if __name__ == '__main__':
    main()
